src/main/scala/mesosphere/marathon/raml/PodStatusConversion.scala

```diff
@@ -143,6 +114,137 @@ trait PodStatusConversion {
+        // not useful to report health conditions for tasks that have never reached a running state
+        None
+      case _ =>
+        val healthy: Option[(Boolean, String)] = maybeContainerSpec.flatMap { containerSpec =>
+          val usingCommandHealthCheck: Boolean = containerSpec.healthCheck.exists(_.command.nonEmpty)
+          if (usingCommandHealthCheck) {
+            Some(status.mesosStatus.withFilter(_.hasHealthy).map(_.getHealthy).fold(false -> HEALTH_UNREPORTED) { h =>
+              h -> HEALTH_REPORTED
+            })
+          } else {
+            val ep = healthCheckEndpoint(containerSpec)
+            ep.map { endpointName =>
+              val epHealthy: Option[Boolean] = endpointStatuses.find(_.name == endpointName).flatMap(_.healthy)
+              // health check endpoint was specified, but if we don't have a value for health yet then generate a
+              // meaningful reason code
+              epHealthy.fold(false -> HEALTH_UNREPORTED) { h => h -> HEALTH_REPORTED }
+            }
+          }
+        }
+        healthy.map { h =>
+          StatusCondition(
+            name = STATUS_CONDITION_HEALTHY,
+            lastChanged = since, // TODO(jdef) pods we should be checking for a real change against some history
+            lastUpdated = since,
+            value = h._1.toString,
+            reason = Some(h._2)
+          )
+        }
+    }
+
+  def endpointStatuses(
+    pod: PodDefinition,
+    maybeContainerSpec: Option[MesosContainer],
+    task: Task): Seq[ContainerEndpointStatus] =
+
+    maybeContainerSpec.flatMap { containerSpec =>
+      task.launched.flatMap { launched =>
+
+        val taskHealthy: Option[Boolean] = // only calculate this once so we do it here
+          launched.status.mesosStatus.withFilter(_.hasHealthy).map(_.getHealthy)
+
+        task.taskId.containerName.flatMap { containerName =>
+          pod.containers.find(_.name == containerName).flatMap { containerSpec =>
+            val endpointRequestedHostPort: Seq[String] = containerSpec.endpoints.collect {
+              case Endpoint(name, _, Some(_), _, _) => name
+            }(collection.breakOut)
+
+            val reservedHostPorts: Seq[Int] = launched.hostPorts
+
+            assume(
+              endpointRequestedHostPort.size == reservedHostPorts.size,
+              s"number of reserved host ports ${reservedHostPorts.size} should equal number of" +
+                s"requested host ports ${endpointRequestedHostPort.size}")
+
+            // we assume that order has been preserved between the allocated port list and the endpoint list
+            // TODO(jdef) pods what actually guarantees that this doesn't change? (do we check this upon pod update?)
+            def reservedEndpointStatus: Map[String, ContainerEndpointStatus] =
+              endpointRequestedHostPort.zip(reservedHostPorts).map { entry =>
```

Comment:
I need to update the `healthy` property of the endpoint that's linked to the health check. thought it would be better to find the endpoint by looking it up from a Map instead of iterating through the sequence. It probably doesn't matter.